tsdb package tests on master are deadlocking

Here's a stacktrace of what I think is the relevant test. I ran it with `-parallel=1 -v` so it should be the correct test.

```
=== RUN   TestStore_BackupRestoreShard
=== RUN   TestStore_BackupRestoreShard/inmem
ts=2018-06-12T16:28:02.272141Z lvl=info msg="Using data dir" log_id=08eD2~O0000 service=store path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009
ts=2018-06-12T16:28:02.272196Z lvl=info msg="Open store (start)" log_id=08eD2~O0000 service=store trace_id=08eD2~O0001 op_name=tsdb_open op_event=start
ts=2018-06-12T16:28:02.272250Z lvl=info msg="Open store (end)" log_id=08eD2~O0000 service=store trace_id=08eD2~O0001 op_name=tsdb_open op_event=end op_elapsed=0.056ms
ts=2018-06-12T16:28:02.272326Z lvl=info msg="Using data dir" log_id=08eD2~O0002 service=store path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188
ts=2018-06-12T16:28:02.272348Z lvl=info msg="Open store (start)" log_id=08eD2~O0002 service=store trace_id=08eD2~O0003 op_name=tsdb_open op_event=start
ts=2018-06-12T16:28:02.272378Z lvl=info msg="Open store (end)" log_id=08eD2~O0002 service=store trace_id=08eD2~O0003 op_name=tsdb_open op_event=end op_elapsed=0.031ms
ts=2018-06-12T16:28:02.311284Z lvl=info msg="tsm1 WAL starting" log_id=08eD2~O0000 engine=tsm1 service=wal segment_size=10485760
ts=2018-06-12T16:28:02.311303Z lvl=info msg="tsm1 WAL writing" log_id=08eD2~O0000 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0/100
ts=2018-06-12T16:28:02.311485Z lvl=info msg="Reloaded WAL cache" log_id=08eD2~O0000 engine=tsm1 path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0/100 duration=0.039ms
ts=2018-06-12T16:28:02.311561Z lvl=info msg="Meta data index for shard loaded" log_id=08eD2~O0000 engine=tsm1 id=100 duration=0.062ms
ts=2018-06-12T16:28:02.325277Z lvl=info msg="Closing WAL file" log_id=08eD2~O0000 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0/100
ts=2018-06-12T16:28:02.325562Z lvl=info msg="Using data dir" log_id=08eD2~aG000 service=store path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009
ts=2018-06-12T16:28:02.325618Z lvl=info msg="Open store (start)" log_id=08eD2~aG000 service=store trace_id=08eD2~aG001 op_name=tsdb_open op_event=start
ts=2018-06-12T16:28:02.327520Z lvl=info msg="tsm1 WAL starting" log_id=08eD2~aG000 engine=tsm1 service=wal segment_size=10485760
ts=2018-06-12T16:28:02.327533Z lvl=info msg="tsm1 WAL writing" log_id=08eD2~aG000 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0/100
ts=2018-06-12T16:28:02.327815Z lvl=info msg="Reading file" log_id=08eD2~aG000 engine=tsm1 service=cacheloader path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0/100/_00001.wal size=60
ts=2018-06-12T16:28:02.327868Z lvl=info msg="Reloaded WAL cache" log_id=08eD2~aG000 engine=tsm1 path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0/100 duration=0.135ms
ts=2018-06-12T16:28:02.328588Z lvl=info msg="Meta data index for shard loaded" log_id=08eD2~aG000 engine=tsm1 id=100 duration=0.703ms
ts=2018-06-12T16:28:02.364003Z lvl=info msg="Opened shard" log_id=08eD2~aG000 service=store trace_id=08eD2~aG001 op_name=tsdb_open index_version=inmem path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/db0/rp0/100 duration=36.846ms
ts=2018-06-12T16:28:02.364039Z lvl=info msg="invalid shard ID found at path" log_id=08eD2~aG000 service=store trace_id=08eD2~aG001 op_name=tsdb_open path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/wal/db0/rp0
ts=2018-06-12T16:28:02.364086Z lvl=info msg="Open store (end)" log_id=08eD2~aG000 service=store trace_id=08eD2~aG001 op_name=tsdb_open op_event=end op_elapsed=38.460ms
ts=2018-06-12T16:28:02.364137Z lvl=info msg="Cache snapshot (start)" log_id=08eD2~aG000 engine=tsm1 trace_id=08eD2~k0000 op_name=tsm1_cache_snapshot op_event=start
ts=2018-06-12T16:28:02.364275Z lvl=info msg="Snapshot for path deduplicated" log_id=08eD2~aG000 engine=tsm1 path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/db0/rp0/100 duration=0.021ms
ts=2018-06-12T16:28:02.374636Z lvl=info msg="Snapshot for path written" log_id=08eD2~aG000 engine=tsm1 trace_id=08eD2~k0000 op_name=tsm1_cache_snapshot path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/db0/rp0/100 duration=10.505ms
ts=2018-06-12T16:28:02.374666Z lvl=info msg="Cache snapshot (end)" log_id=08eD2~aG000 engine=tsm1 trace_id=08eD2~k0000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=10.533ms
ts=2018-06-12T16:28:02.374678Z lvl=info msg="Creating snapshot" log_id=08eD2~aG000 engine=tsm1 service=filestore dir=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-514144009/db0/rp0/100
ts=2018-06-12T16:28:02.407245Z lvl=info msg="tsm1 WAL starting" log_id=08eD2~O0002 engine=tsm1 service=wal segment_size=10485760
ts=2018-06-12T16:28:02.407265Z lvl=info msg="tsm1 WAL writing" log_id=08eD2~O0002 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/wal/db0/rp0/100
ts=2018-06-12T16:28:02.407488Z lvl=info msg="Reloaded WAL cache" log_id=08eD2~O0002 engine=tsm1 path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/wal/db0/rp0/100 duration=0.051ms
ts=2018-06-12T16:28:02.407593Z lvl=info msg="Meta data index for shard loaded" log_id=08eD2~O0002 engine=tsm1 id=100 duration=0.082ms
ts=2018-06-12T16:28:02.418547Z lvl=info msg="Closing WAL file" log_id=08eD2~O0002 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/wal/db0/rp0/100
ts=2018-06-12T16:28:02.418791Z lvl=info msg="tsm1 WAL starting" log_id=08eD2~O0002 engine=tsm1 service=wal segment_size=10485760
ts=2018-06-12T16:28:02.418818Z lvl=info msg="tsm1 WAL writing" log_id=08eD2~O0002 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/wal/db0/rp0/100
ts=2018-06-12T16:28:02.419333Z lvl=info msg="Opened file" log_id=08eD2~O0002 engine=tsm1 service=filestore path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/db0/rp0/100/000000001-000000001.tsm id=0 duration=0.130ms
ts=2018-06-12T16:28:02.419441Z lvl=info msg="Reloaded WAL cache" log_id=08eD2~O0002 engine=tsm1 path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/wal/db0/rp0/100 duration=0.068ms
ts=2018-06-12T16:28:02.429262Z lvl=info msg="Meta data index for shard loaded" log_id=08eD2~O0002 engine=tsm1 id=100 duration=9.805ms
ts=2018-06-12T16:28:02.429481Z lvl=info msg="Closing WAL file" log_id=08eD2~O0002 engine=tsm1 service=wal path=/var/folders/w5/25fg3zv56y7cgd20q4fx_24m0000gn/T/influxdb-tsdb-811519188/wal/db0/rp0/100
panic: test timed out after 5m0s

goroutine 61190 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:1240 +0xfc
created by time.goFunc
	/usr/local/Cellar/go/1.10.2/libexec/src/time/sleep.go:172 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc47fba2ff0, 0x176f087, 0x1c, 0x1786c18, 0x1078301)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc420190000)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc420190000, 0xc420087df8)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
testing.runTests(0xc420134e60, 0x1b7ab20, 0x51, 0x51, 0x1012589)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc4200d6a00, 0x0)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:978 +0x171
main.main()
	_testmain.go:324 +0x151

goroutine 61222 [chan receive]:
testing.(*T).Parallel(0xc47fba2690)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_DeleteRetentionPolicy(0xc47fba2690)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:34 +0x2f
testing.tRunner(0xc47fba2690, 0x1786c58)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 39977 [select]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compact(0xc42a3ea120, 0xc429080bc0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1927 +0x27f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableLevelCompactions.func1(0xc429080bc0, 0xc42a3ea120)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:396 +0x5b
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableLevelCompactions
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:396 +0x145

goroutine 61229 [chan receive]:
testing.(*T).Parallel(0xc47fba2d20)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_Open_InvalidRetentionPolicy(0xc47fba2d20)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:351 +0x2f
testing.tRunner(0xc47fba2d20, 0x1786c90)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 39976 [select]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactCache(0xc42a3ea120)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1880 +0x13f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions.func1(0xc429080bb0, 0xc42a3ea120)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:475 +0x51
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:475 +0x137

goroutine 373 [select]:
github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start.func2(0xc420a7efc0, 0xc4209cb4e0, 0xc4209cb4c0, 0xc420a2f630)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:87 +0x193
created by github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:84 +0x14f

goroutine 473 [select]:
github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start.func2(0xc42060d0e0, 0xc4204a9960, 0xc4204a9940, 0xc420465400)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:87 +0x193
created by github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:84 +0x14f

goroutine 411 [select]:
github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start.func2(0xc42076f140, 0xc42065ec40, 0xc42065ec20, 0xc420610d70)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:87 +0x193
created by github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:84 +0x14f

goroutine 61224 [chan receive]:
testing.(*T).Parallel(0xc47fba2870)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_DeleteSeries_NonExistentDB(0xc47fba2870)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:147 +0x2f
testing.tRunner(0xc47fba2870, 0x1786c60)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 40000 [select]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compact(0xc42a3ea480, 0xc4290812f0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1927 +0x27f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableLevelCompactions.func1(0xc4290812f0, 0xc42a3ea480)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:396 +0x5b
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableLevelCompactions
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:396 +0x145

goroutine 59277 [select]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactCache(0xc469f1d320)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1880 +0x13f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions.func1(0xc47bbb3e00, 0xc469f1d320)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:475 +0x51
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:475 +0x137

goroutine 476 [select]:
github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start.func2(0xc42060d1a0, 0xc4204a99c0, 0xc4204a99a0, 0xc4204654a0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:87 +0x193
created by github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:84 +0x14f

goroutine 470 [select]:
github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start.func2(0xc42060cfc0, 0xc4204a9900, 0xc4204a98e0, 0xc4204653b0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:87 +0x193
created by github.com/influxdata/influxdb/tsdb.(*PointBatcher).Start
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/batcher.go:84 +0x14f

goroutine 59278 [select]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compact(0xc469f1d320, 0xc47bbb3e10)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1927 +0x27f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableLevelCompactions.func1(0xc47bbb3e10, 0xc469f1d320)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:396 +0x5b
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableLevelCompactions
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:396 +0x145

goroutine 61233 [chan receive]:
github.com/influxdata/influxdb/tsdb.(*Store).walkShards(0xc4814f65a0, 0xc471e73918, 0x1, 0x1, 0x17864f0, 0x17e9880, 0xc420078040)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store.go:889 +0xf1
github.com/influxdata/influxdb/tsdb.(*Store).Close(0xc4814f65a0, 0x1787af8, 0xc481010fa0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store.go:421 +0xa8
github.com/influxdata/influxdb/tsdb_test.(*Store).Close(0xc47fb99e80, 0x0, 0x0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:1792 +0x7b
github.com/influxdata/influxdb/tsdb_test.TestStore_BackupRestoreShard.func1(0x1762814, 0x5)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:560 +0xad9
github.com/influxdata/influxdb/tsdb_test.TestStore_BackupRestoreShard.func2(0xc47fba30e0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:568 +0x3d
testing.tRunner(0xc47fba30e0, 0xc47fb99e40)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 39999 [select]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactCache(0xc42a3ea480)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1880 +0x13f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions.func1(0xc4290812e0, 0xc42a3ea480)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:475 +0x51
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:475 +0x137

goroutine 61232 [chan receive]:
testing.(*T).Run(0xc47fba30e0, 0x1762814, 0x5, 0xc47fb99e40, 0x10e362d)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:825 +0x301
github.com/influxdata/influxdb/tsdb_test.TestStore_BackupRestoreShard(0xc47fba2ff0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:567 +0xb5
testing.tRunner(0xc47fba2ff0, 0x1786c18)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61225 [chan receive]:
testing.(*T).Parallel(0xc47fba2960)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_DeleteShard(0xc47fba2960)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:165 +0x2f
testing.tRunner(0xc47fba2960, 0x1786c70)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61228 [chan receive]:
testing.(*T).Parallel(0xc47fba2c30)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_Open_InvalidDatabaseFile(0xc47fba2c30)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:325 +0x2f
testing.tRunner(0xc47fba2c30, 0x1786c88)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61227 [chan receive]:
testing.(*T).Parallel(0xc47fba2b40)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_Open(0xc47fba2b40)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:282 +0x2f
testing.tRunner(0xc47fba2b40, 0x1786ca0)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61314 [semacquire]:
sync.runtime_Semacquire(0xc48136e4e4)
	/usr/local/Cellar/go/1.10.2/libexec/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc48136e4d8)
	/usr/local/Cellar/go/1.10.2/libexec/src/sync/waitgroup.go:129 +0x72
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*TSMReader).Close(0xc48136e4d0, 0x0, 0x0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:409 +0x3d
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Close(0xc4817aa410, 0x1787ff0, 0xc480636b40)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:597 +0xdd
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).Close(0xc480636b40, 0x0, 0x0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:744 +0xa5
github.com/influxdata/influxdb/tsdb.(*Shard).close(0xc422ef6d80, 0x1787ff0, 0xc422ef6f20)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/shard.go:365 +0x53
github.com/influxdata/influxdb/tsdb.(*Shard).Close(0xc422ef6d80, 0x0, 0x0)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/shard.go:355 +0x79
github.com/influxdata/influxdb/tsdb.(*Store).Close.func1(0xc422ef6d80, 0x1787ff8, 0xc4202f1788)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store.go:422 +0x2b
github.com/influxdata/influxdb/tsdb.(*Store).walkShards.func1(0x17864f0, 0xc4820e8120, 0xc422ef6d80)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store.go:878 +0x48
created by github.com/influxdata/influxdb/tsdb.(*Store).walkShards
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store.go:877 +0x80

goroutine 61231 [chan receive]:
testing.(*T).Parallel(0xc47fba2f00)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestShards_CreateIterator(0xc47fba2f00)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:411 +0x2f
testing.tRunner(0xc47fba2f00, 0x1786bf0)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61230 [chan receive]:
testing.(*T).Parallel(0xc47fba2e10)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_Open_InvalidShard(0xc47fba2e10)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:381 +0x2f
testing.tRunner(0xc47fba2e10, 0x1786c98)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61226 [chan receive]:
testing.(*T).Parallel(0xc47fba2a50)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_CreateShardSnapShot(0xc47fba2a50)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:254 +0x2f
testing.tRunner(0xc47fba2a50, 0x1786c48)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0

goroutine 61223 [chan receive]:
testing.(*T).Parallel(0xc47fba2780)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:697 +0x205
github.com/influxdata/influxdb/tsdb_test.TestStore_CreateShard(0xc47fba2780)
	/Users/jsternberg/go/src/github.com/influxdata/influxdb/tsdb/store_test.go:110 +0x2f
testing.tRunner(0xc47fba2780, 0x1786c50)
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.2/libexec/src/testing/testing.go:824 +0x2e0
FAIL	github.com/influxdata/influxdb/tsdb	300.082s
```

--- a/cmd/influxd/backup/backup.go
+++ b/cmd/influxd/backup/backup.go
@@ -19,7 +19,7 @@ import (
 	"github.com/influxdata/influxdb/cmd/influxd/backup_util"
 	"github.com/influxdata/influxdb/services/snapshotter"
 	"github.com/influxdata/influxdb/tcp"
-        gzip "github.com/klauspost/pgzip"
+	gzip "github.com/klauspost/pgzip"
 )
 
 const (

--- a/tsdb/store_test.go
+++ b/tsdb/store_test.go
@@ -533,6 +533,7 @@ func TestStore_BackupRestoreShard(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
+		defer itr.Close()
 		fitr := itr.(query.FloatIterator)
 
 		// Read values from iterator. The host=serverA points should come first.

